tests/functional/n/non_str_assignment_to_dunder_name.py

```diff
@@ -0,0 +1,18 @@
+# pylint: disable=missing-module-docstring, missing-class-docstring
+# pylint: disable=too-few-public-methods, missing-function-docstring
+
+
+class ExampleClass():
+    pass
+
+
+def example_function():
+    pass
+
+
+ExampleClass.__name__ = 1  # [non-str-assignment-to-dunder-name]
```

Comment:
Let's add a couple of additional test cases:

* One with a function call on the RHS returning a string (should not emit)
* One with a function call on the RHS returning an integer (should emit)
* One with a functional on the RHS returning multiple types (should not emit)
* Do `from unknown import Unknown` and place `Unknown` on the RHS (should not emit)

Reply:
Thank you!

I wasn't quite sure what 

> One with a functional on the RHS returning multiple types (should not emit)

meant, but I added a return of a tuple, and a return that coule be multiple types based on runtime conditions. Hope that covers it!

Changes are in https://github.com/PyCQA/pylint/pull/3271/commits/b4f795582bf924384517faa46aa89dc62fc5b086